src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -128,6 +128,20 @@ internal sealed class MemberSignatureComparer : IEqualityComparer<Symbol>
             considerRefKindDifferences: true,
             typeComparison: TypeCompareKind.AllIgnoreOptions);
 
+        /// <summary>
+        /// This instance is used to determine if an interceptor can be applied to an interceptable method.
+        /// NB: when a classic extension method is intercepting an instance method call, a normalization to 'ReducedExtensionMethodSymbol' must be performed first.
+        /// </summary>
+        public static readonly MemberSignatureComparer InterceptorsComparer = new MemberSignatureComparer(
+            considerName: false,
+            considerExplicitlyImplementedInterfaces: false,
+            considerReturnType: true,
+            considerTypeConstraints: false,
+            considerCallingConvention: false,
+            considerRefKindDifferences: true,
+            considerArity: false,
+            typeComparison: TypeCompareKind.ObliviousNullableModifierMatchesAny);
```

Comment:
Spec (not blocking this PR): I think we should use the same type comparison that we do for OHI (see `CSharpOverrideComparer` below). So `dynamic`/`object` difference would be okay and we'd only warn for nullability differences.

Reply:
I'm open to this. Noting the question in the spec.